Bottom: 47e54029ad559cdf565b84ba7732225a32190799
Top:    457f5e25f826e765c4650ceb9fa49ce37ee816b2
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-04-24 13:47:08 +0800

Refresh of tuntap-enable-premmption-early

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 28cfa64..36888fa 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1686,6 +1686,8 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 			goto err_xdp;
 		}
 	}
+	rcu_read_unlock();
+	preempt_enable();
 
 	skb = build_skb(buf, buflen);
 	if (!skb) {
@@ -1707,9 +1709,6 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 		return NULL;
 	}
 
-	rcu_read_unlock();
-	preempt_enable();
-
 	return skb;
 
 err_redirect:
